src/lsstseries/ensemble.py

```diff
@@ -294,14 +364,73 @@ def from_parquet(
         else:
             columns = [self._time_col, self._flux_col, self._err_col, self._band_col]
 
-        # Read in a parquet file
-        self._data = dd.read_parquet(file, index=self._id_col, columns=columns, split_row_groups=True)
+        # Read in the source parquet file(s)
+        self._source = dd.read_parquet(
+            source_file, index=self._id_col, columns=columns, split_row_groups=True
+        )
 
         if npartitions and npartitions > 1:
-            self._data = self._data.repartition(npartitions=npartitions)
+            self._source = self._source.repartition(npartitions=npartitions)
         elif partition_size:
-            self._data = self._data.repartition(partition_size=partition_size)
+            self._source = self._source.repartition(partition_size=partition_size)
+
+        if object_file:  # read from parquet files
+            # Read in the object file
+            file_table = dd.read_parquet(object_file, index=self._id_col, split_row_groups=True)
+
+            # Generate an object table from the source table, then merge
+            generated = self._generate_object_table()
+            self._object = file_table.merge(generated, how="right", on=[self._id_col])
```

Comment:
I'm in the same boat, where I'd think we want to explicitly no hold objects that have no sources, as they aren't timeseries, but am aware I may be out of touch with what a user would want. @nevencaplar would you ever want to store an object that has no sources? If we drop an object, then it can always be re-generated when sources are added. Also, I don't quite understand the last question, what do you mean by filtering sources not in the observation file?

Reply:
Hm, that is a good question. I think that in general we do want to hold objects which have no sources, i.e., if you filter out sources, I think you still want to have the memory that there was an object before filtering. Or, more pertinent  to this case where we are loading data, you would want to be able to identify that something is weird. How it is written now, if I understand the discussion, you might have a problem in the data and you would not be able to notice it!

Reply:
Good to know. Unless there's a strong objection, I'll capture this in an issue. The reason being is that as the code is currently written the object table is derivative of the source table. Having the object table hold an independent list of objects not found in source will need some additional thinking.